File: glossier/exercise/ruby/test/fixtures.rb
module Exercise
  module Fixtures
    def self.milky_jelly_cleanser
      Product.new(
        category: "skincare",
        id: 101,
        name: "Milky Jelly Cleanser"
      )
    end

    def self.moisturizing_moon_mask
      Product.new(
        category: "skincare",
        id: 128,
        name: "Moisturizing Moon Mask"
      )
    end

    def self.invisible_shield
      Product.new(
        category: "skincare",
        id: 122,
        name: "Invisible Shield"
      )
    end

    def self.haloscope
      Product.new(
        category: "makeup",
        id: 395,
        name: "Haloscope"
      )
    end

    def self.glossier_you
      Product.new(
        category: "fragrance",
        id: 910,
        name: "Glossier You"
      )
    end

    def self.new_york
      Location.new(
        city: "New York City",
        country: "USA",
        id: 1,
        state: "New York"
      )
    end

    def self.montreal
      Location.new(
        city: "Montreal",
        country: "Canada",
        id: 2,
        state: "Quebec"
      )
    end

    def self.san_francisco
      Location.new(
        city: "San Francisco",
        country: "USA",
        id: 3,
        state: "California"
      )
    end

    def self.los_angeles
      Location.new(
        city: "Los Angeles",
        country: "USA",
        id: 4,
        state: "California"
      )
    end

    def self.locations
      [new_york, montreal, san_francisco, los_angeles]
    end
  end
end

File: glossier/exercise/javascript/test/fixtures.js
var exports = module.exports = {}

const Location = require('lib/exercise/models/location').Location
const Product = require('lib/exercise/models/product').Product

exports.milkyJellyCleanser = new Product('skincare', 101, 'Milky Jelly Cleanser')
exports.moisturizingMoonMask = new Product('skincare', 128, 'Milky Jelly Cleanser')
exports.invisibleShield = new Product('skincare', 122, 'Invisible Shield')
exports.haloscope = new Product('makeup', 395, 'Haloscope')
exports.glossierYou = new Product('fragrance', 910, 'Glossier You')

exports.newYork = new Location('New York City', 'USA', 1, 'New York')
exports.montreal = new Location('Montreal', 'Canada', 2, 'Quebec')
exports.sanFrancisco = new Location('San Francisco', 'USA', 3, 'California')
exports.losAngeles = new Location('Los Angeles', 'USA', 4, 'California')

exports.locations = [
  exports.newYork,
  exports.montreal,
  exports.sanFrancisco,
  exports.losAngeles
]
